src/protocols/multisig/mod.rs

```diff
@@ -0,0 +1,157 @@
+/*
+    Multisig Schnorr
+
+    Copyright 2018 by Kzen Networks
+
+    This file is part of Multisig Schnorr library
+    (https://github.com/KZen-networks/multisig-schnorr)
+
+    Multisig Schnorr is free software: you can redistribute
+    it and/or modify it under the terms of the GNU General Public
+    License as published by the Free Software Foundation, either
+    version 3 of the License, or (at your option) any later version.
+
+    @license GPL-3.0+ <https://github.com/KZen-networks/multisig-schnorr/blob/master/LICENSE>
+*/
+
+//! Schnorr {n,n}-Signatures based on Accountable-Subgroup Multisignatures
+//!
+//See (https://pdfs.semanticscholar.org/6bf4/f9450e7a8e31c106a8670b961de4735589cf.pdf)
+use cryptography_utils::elliptic::curves::traits::*;
+use cryptography_utils::{BigInt, FE, GE};
+
+use cryptography_utils::cryptographic_primitives::hashing::hash_sha256::HSha256;
+use cryptography_utils::cryptographic_primitives::hashing::traits::*;
+use protocols::multisig;
+#[derive(Debug, Clone)]
+pub struct Keys {
+    pub I: KeyPair,
+    pub X: KeyPair,
+}
+
+#[derive(Debug, Clone)]
+pub struct KeyPair {
+    pub public_key: GE,
+    private_key: FE,
+}
+
+impl KeyPair {
+    pub fn create() -> KeyPair {
+        let ec_point: GE = ECPoint::generator();
+        let private_key: FE = ECScalar::new_random();
+        let public_key = ec_point.scalar_mul(&private_key.get_element());
+        KeyPair {
+            public_key,
+            private_key,
+        }
+    }
+
+    pub fn create_from_private_key(private_key: &BigInt) -> KeyPair {
+        let ec_point: GE = ECPoint::generator();
+        let private_key: FE = ECScalar::from(private_key);
+        let public_key = ec_point.scalar_mul(&private_key.get_element());
+        KeyPair {
+            public_key,
+            private_key,
+        }
+    }
+}
+
+impl Keys {
+    //TODO:: add create from private key
+    pub fn create() -> Keys {
+        let I = KeyPair::create();
+        let X = KeyPair::create();
+        Keys { I, X }
+    }
+
+    pub fn broadcast(keys: &Keys) -> Vec<&GE> {
+        return vec![&keys.I.public_key, &keys.X.public_key];
+    }
+
+    pub fn collect_and_compute_challenge(ix_vec: &Vec<Vec<&GE>>) -> FE {
+        let new_vec: Vec<&GE> = Vec::new();
+        let concat_vec = ix_vec.iter().fold(new_vec, |mut acc, x| {
```

Comment:
`return acc` -> `acc`

P.S.
Consider using [flatten](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.flatten) here, if possible.

Reply:
`flatten` is nice. Problem is that we need to implement `from_iter` for points. I am not sure if this is possible.